sim/src/main/scala/midasexamples/AutoCounterModule.scala

```diff
@@ -27,6 +27,11 @@ class AutoCounterModuleDUT(
 
   PerfCounter(enabled4, "ENABLED_DIV_4", "Count the number of times the enabled cycle count is divisible by 4. Should be equal to number of cycles minus reset cycles divided by 4")
 
+  // Multibit event
```

Comment:
FWIW, this is the test.

Reply:
Oh.  Cool.   I wasn't sure if it was or not because I didn't see something checking the counted value against some expectation.  I need to learn more about how the tests work. :)

Reply:
How these particular tests work is by diffing the AutoCounter-generated output against an printf that's been formatted to match that output (see below). It's a little bit hacky, but at least it's simple.

Reply:
I was sort of seeing it before but didn't see anything that was parsing the output of the print statements and making sure they were the same, so I (very incorrectly) assumed that this wasn't one of the CI tests but was something that could be visually inspected for correctness.  

In other words, I was forgetting the leap back into 
https://github.com/firesim/firesim/blob/fec239847ac57c7b13b1782ba933b9fc91de9121/sim/src/test/scala/midasexamples/TutorialSuite.scala#L110-L123  
where `diffSynthesizedLog` is being called.  I was also being thrown off by the 'test' code living in 'src/main' instead of 'src/test'.

Not trying to complain about the tests being hacky.  It's great that they are there.  I just forget about how they are plumbed together because it goes through a few layers of SBT, makefile, SBT and I get confused.

Reply:
One way to rationalize how this works, is that these duts are targets with their own generators. The tests merely consist of a means to invoke parts of the make-based flow and diff generated outputs. It started off this way because the midasexamples were suppose to a simple way to demonstrate how you could use MIDAS not a target that wasn't rocket-chip based. I've since added a bunch more targets that both demonstrate simple features but increasingly just act as tests. It probably makes sense to pull them into src/test, and with a little SBT chicanery everything will work the same way. 




The tests are wrappers consist of wrappers around existing

Reply:
While it might be possible to reorganize things to put these into 'src/test', I think that a less drastic way to aleviate what I'm describing would be adding Scaladoc for `AutoCounterModule`, `AutoCounterCoverModule` and `AutoCounterPrintfModule` that captures what you just wrote: 'this is a  target its their own generator that demonstrates XXX and is suitable for use in testing by extending the abstract `TutorialSuite`'.

I was going to suggest a single file-scope Scaladoc but looking more closely, that doesn't seem to be a thing in Scaladoc as it is in C++ Doxygen.  I'll see if I can propose them as comments on this PR in the appropriate places.

Reply:
Grr.  Of course I can't.  The lines I want to propose patches for are too far away from your original patches so why the heck would I ever want to comment on them!  Thanks Github!!!

I'll submit my suggestions as another PR into this PR (rather than doing nothing or pushing them straight into your branch).